mitiq/tests/test_measurement_result.py

```diff
@@ -69,16 +71,69 @@ def test_empty():
     assert result.result == []
 
 
-def test_iter():
-    bitstrings = [[0, 1, 1], [0, 1, 0], [1, 0, 0]]
-
-    for m, bits in zip(MeasurementResult(bitstrings), bitstrings):
-        assert m == bits
-
-
 def test_convert_to_array():
     bitstrings = [[0, 0], [0, 1], [1, 0]]
     result = MeasurementResult(bitstrings)
     assert np.allclose(result.asarray, np.array(bitstrings))
 
     assert np.allclose(MeasurementResult([]).asarray, np.array([]))
+
+
+@pytest.mark.parametrize("qubit_indices", ((0, 1), (1, 20)))
+def test_measurement_result_with_strings(qubit_indices):
+    """Try using strings instead of lists of integers."""
+    bitstrings = ["00", "01", "10"]
+    int_bitstrings = [[0, 0], [0, 1], [1, 0]]
+
+    result = MeasurementResult(bitstrings, qubit_indices=qubit_indices)
+
+    assert result.nqubits == 2
+    assert result.qubit_indices == qubit_indices
+    assert result.shots == 3
+    assert np.allclose(result.result, int_bitstrings)
```

Comment:
Does exact equality not hold? I would think it should, as `result.result` should be a list of ints.

Reply:
Equality works!